Updated docs for the test refactor (Tests PR 2/3) (#1835)
ManimCommunity/manim: docs/source/contributing/testing.rst

``manim/mobject/geometry.py``, add the CircleTest to
``test/test_geometry.py``.

> You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** take as a parameter a variable named ``scene`` that will be used like ``self`` in a standard ``construct`` method.

Reviewer: This seems important enough that it should be an admonition.

+a clarity suggestion
```suggestion
.. important::
    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.
```

Example : in ``test_geometry.py`` :

> .. code:: python
>   @frames_comparison

Reviewer: currently isn't showing up in the docs, needs a newline
```suggestion
.. code:: python

  @frames_comparison
```

  def test_circle(scene):
      circle = Circle()
      scene.play(Animation(circle))

The decorator can be used with or without parentheses. **By default, the test only tests the last frame. To enable multi-frame testing, you have to set ``last_frame=False`` in the parameters.**.

> .. code:: python
>   @frames_comparison(last_frame=False)

Reviewer: same as above
```suggestion
.. code:: python

  @frames_comparison(last_frame=False)
```

  def test_circle(scene):
      circle = Circle()
      scene.play(Animation(circle))

You can also specify, when needed, which base scene you need (ThreeDScene, for example) : 

.. code:: python

>   @frames_comparison(last_frame=False, base_scene=ThreeDScene)

Reviewer: same as above
```suggestion

  @frames_comparison(last_frame=False, base_scene=ThreeDScene)
```

  def test_circle(scene):
      circle = Circle()
      scene.play(Animation(circle))

Feel free to check the documentation of ``@frames_comparison`` for more.

Note that tests name must follow the syntax ``test_<thing_to_test>``, otherwise pytest will not recognize it as a test.

.. warning::
  If you run pytest now, you will get a ``FileNotFound`` error. This is because
  you have not created control data for your test. 

To create the control data for your test, you have to use the flag ``--set_test`` along with pytest. 

> For the example above, it would be ``pytest test_geometry.py::test_circle --set_test -s``. (``-s`` is here to see manim logs, so you can see what's going on). 

Reviewer: I think having having a `.. code-block::bash` with the input needed would make this more explicit/noticeable

Author: Will do, thanks

``manim/mobject/geometry.py``, add the CircleTest to
``test/test_geometry.py``.

> The name of the module is indicated by the variable __module_test__, that **must** be decalared in any graphical test file. The module name is used to store the graphical control data.

Reviewer: ```suggestion
The name of the module is indicated by the variable __module_test__, that **must** be declared in any graphical test file. The module name is used to store the graphical control data.
```

.. important::
    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.

Example : in ``test_geometry.py`` :

> .. code:: python
>   from tests.test_graphical_units.testing.frames_comparison import frames_comparison

Reviewer: A space is needed. Also for the sake of having one complete example, all the imports + `__module_test__`.
```suggestion
.. code:: python
  
  from manim import *
  from tests.test_graphical_units.testing.frames_comparison import frames_comparison

  __module_test__ = "geometry"
  
```

  @frames_comparison
  def test_circle(scene):
      circle = Circle()
      scene.play(Animation(circle))

> The decorator can be used with or without parentheses. **By default, the test only tests the last frame. To enable multi-frame testing, you have to set ``last_frame=False`` in the parameters.**.

Reviewer: Removed double period.
```suggestion
The decorator can be used with or without parentheses. **By default, the test only tests the last frame. To enable multi-frame testing, you have to set ``last_frame=False`` in the parameters.**
```

.. code:: python

  @frames_comparison(last_frame=False)
  def test_circle(scene):
      circle = Circle()
      scene.play(Animation(circle))

You can also specify, when needed, which base scene you need (ThreeDScene, for example) : 

.. code:: python

  @frames_comparison(last_frame=False, base_scene=ThreeDScene)
  def test_circle(scene):
      circle = Circle()
      scene.play(Animation(circle))

Feel free to check the documentation of ``@frames_comparison`` for more.

Note that tests name must follow the syntax ``test_<thing_to_test>``, otherwise pytest will not recognize it as a test.

.. warning::
  If you run pytest now, you will get a ``FileNotFound`` error. This is because
  you have not created control data for your test. 

To create the control data for your test, you have to use the flag ``--set_test`` along with pytest. 
For the example above, it would be 

> .. code-block::bash

Reviewer: a space is needed here
```suggestion
.. code-block:: bash
```

``manim/mobject/geometry.py``, add the CircleTest to
``test/test_geometry.py``.

The name of the module is indicated by the variable __module_test__, that **must** be declared in any graphical test file. The module name is used to store the graphical control data.

.. important::
    You will need to use the ``frames_comparison`` decorator to create a test. The test function **must** accept a 
    parameter named ``scene`` that will be used like ``self`` in a standard ``construct`` method.

> Example : in ``test_geometry.py`` :

Reviewer: ```suggestion
Here's an example in ``test_geometry.py``:
```

> .. code:: python

Reviewer: ```suggestion
.. code:: python

```
